Source/Core/VideoBackends/OGL/TextureCache.h

```diff
@@ -20,43 +24,26 @@ class TextureCache : public TextureCacheBase
   TextureCache();
   ~TextureCache();
 
-  static void DisableStage(unsigned int stage);
-  static void SetStage();
+  static TextureCache* GetInstance();
 
   bool SupportsGPUTextureDecode(TextureFormat format, TlutFormat palette_format) override;
-  void DecodeTextureOnGPU(TCacheEntryBase* entry, u32 dst_level, const u8* data, size_t data_size,
+  void DecodeTextureOnGPU(TCacheEntry* entry, u32 dst_level, const u8* data, size_t data_size,
                           TextureFormat format, u32 width, u32 height, u32 aligned_width,
                           u32 aligned_height, u32 row_stride, const u8* palette,
                           TlutFormat palette_format) override;
 
-private:
-  struct TCacheEntry : TCacheEntryBase
-  {
-    GLuint texture;
-    GLuint framebuffer;
-
-    // TexMode0 mode; // current filter and clamp modes that texture is set to
-    // TexMode1 mode1; // current filter and clamp modes that texture is set to
-
-    TCacheEntry(const TCacheEntryConfig& config);
-    ~TCacheEntry();
-
-    void CopyRectangleFromTexture(const TCacheEntryBase* source,
-                                  const MathUtil::Rectangle<int>& srcrect,
-                                  const MathUtil::Rectangle<int>& dstrect) override;
+  SHADER GetColorCopyProgram() const;
```

Comment:
What do you think about SHADER&? Right now, they are usually quite simple, but this may differ....